### ISSUE TITLE
Incremental Search - Feature request: Add search direction option with All|Up|Down

### ISSUE THREAD

**Issue by martbase** [notepad-plus-plus/notepad-plus-plus#526]
I have come to love using the increamental search feature as I can easily access it on the status bar across many documents while working on multiple documents. 

However since the search direction option is missing sometimes the search wraps around in a long document requiring users to keep track of the line number to know they have reached the end and started over again.

**Comment by tonyk24** [comment]
I have a question regarding the All option. I suppose when you search using Up, it will search until top is reached and then stop, and in the same way, you select Down it will search to end and then stop.

The All option I suppose would search down incrementally, but won't stop at the last line. But what if you want to incrementally search upwards and still continue searching after the first line,, then you have to change the option from from Up to All then.

A solution would be to simply mark when end or beginning of document has been reached, in the status bar for example, and then continue to the next search, like it is done in Office suites.

Another question is what you mean with "I can easily access it on the status bar across many documents while working on multiple documents.". As I can't access Find dialog from status bar.

**Comment by tonyk24** [comment]
Never mind, Found the search from status bar pushing CRL-ALT-I. Still don't understand this feature, since there is an option both backwards, and forward. It also shows when beginning or end of document is reached.

**Comment by martbase** [comment]
The way incremental search is implemented now covers the All direction as it wraps around when it reaches either the top/bottom when searching with the < and > buttons or when you keep pressing Ctrl+Alt+I.
The issue is a user must must keep track of the status shown when either the beginning or end of the document is reached thus the need for an option for specifying the direction to have it stop in addition to indicating when it reaches the start/end.
In the find dialog the three search methods are covered with Direction(Up/Down) and the Wrap around checkbox options.

**Comment by martbase** [comment]
Yes is simpler but it doesn't address the issue of stopping when the last search item is reached (either up/down) hence the user has to manually track the line number to know they have checked all found items otherwise they will waste time going round and round (when not monitoring the current line)

How about adding a checkbox for allowing wrap around when necessary or disabling the behavior?
